### PR TITLE
Use .jar files from the .NET runtime pack

### PR DESCRIPTION
**Pull request by simonrozsival**
Since https://github.com/dotnet/runtime/pull/77386 has been merged, .NET will require a certain class from  `libSystem.Security.Cryptography.Native.Android.jar` that will be located in the runtime pack files.

I'm not sure if this is the best way to pull the .jar from the runtimepack so feedback is welcome.

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets [hunk 1]
```diff
@@ -1341,6 +1341,12 @@ because xbuild doesn't support framework reference assemblies.
   <MakeDir Directories="$(MonoAndroidIntermediateAssemblyDir)" Condition=" !Exists ('$(MonoAndroidIntermediateAssemblyDir)') " />
 </Target>
 
+<Target Name="_CollectRuntimePackJarFiles">
+  <ItemGroup>
+    <AndroidJavaLibrary Include="@(ResolvedFileToPublish)" Condition=" '%(ResolvedFileToPublish.Extension)' == '.jar' " />
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:1346]: Is the `.jar` file in all 4 runtime packs? `android-arm`, `android-arm64`, `android-x86`, android-x64`?

We'll probably have to deduplicate them somehow. Let's wait to look into this, when the changes in dotnet/runtime flow to us.

Comment (simonrozsival) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:1346]: Yes, each runtimepack should have a copy of the jar file.

**Comment by jonathanpeppers** [comment]
I'm going to merge this into our latest Maestro bump, and I'll takeover from there. Thanks!

**Comment by simonrozsival** [comment]
@jonathanpeppers thanks!

**Pull request merged**